W12-Revision.md

```diff
@@ -795,149 +795,137 @@
-        
-        - Cohesion: Are functions related to each other
+<details>
+	<summary> Show answer </summary>	
+
+	Important design characteristics in code design are 
+	
+	Coherence: Do the subcomponents in our logical structure contain the components they are meant to contain?
+	
+	Cohesion: Are the elements and components related to each other?
+	
+	Consistency: Do variable names, functions and responsibilities maintain a consistent order?
+	
+	Coupling: Do components and elements relate strongly with each other (Does altering one piece of code affect the 
+	other)? We want to try and aim for low coupling.
+	
+	Extensibility: How easy is it to add new functions without affecting old code?
+	
+	Flexibility: Can our logical structure be changed without causing any faults
+	
+	Generality: Are our subcomponents in our logical structure defined generally in our structure?
+	
+	Interdependency: Which components in our logical structures are needed to communicate with other components?
+	
+	Seperation of concerns: Are the components in our logical structure distinctly identifiable with each other?
+	
+	Simplicity: Does the solution to create the logical structure make sense and contains no unnecessary and redundant code?
+
+</details>
 
-        - Consistency: Are they consistent
-        
-        - Coupling: How strong the elements are - to what extends does changing one will affect the others.
 
-        - Extensibility: Easy to extend to new functions?
-
-        - Generality: Is the way the program work general as possible
-
-        - Interdependency: what portion of the processing involves interaction between different elements.
-
-        - Seperation of concerns: are internal elements responsible for distinct parts of the system's operation
-
-        - Simplicity
-
-        We want:
-        - Hight extensibility
-        - High reusability
-
-        Can achieve by refactoring following design principles.
-    </details>
-
-
-  
 - What are some of the “design smells”, the blocks to reusability, extensibility, modifiability --- don’t memorise them all but know how to describe them and why they are a problem. 
-    <details>
-    <summary>Show answer</summary>
-
-        Rigidity: Affect modifiability, the system is hard to change, because every changes will lead to more change. (high coupling)
-
-        Fragility: Affect modifiability, the system will break in multiple places.
 
-        Immobilitiy: Affect reusability, the parts could be reuse but too much risks in seperating it.
+<details>
+	<summary> Show answer </summary>	
+
+	Design smells that may be present in poor code include:
+	
+	Rigidity: The system is hard to change because it will cause other parts of 
+		  the system to also change. (An example of high coupling)
+	
+	Fragility: The system may break in multiple places due to the change of one part
+	
+	Immobility: The class is too dependent on the application and can not be run on other environments
```

Comment:
According to the lecture note, immobility they emphasize the fact that "parts could be reused in other systems" I think this is important. 

The rest is fine